Adding documentation on Product Filter. (#50255)
ansible/ansible: docs/docsite/rst/user_guide/playbooks_filters.rst

Also see the :ref:`zip_filter`

Product Filters
```````````````

> The product filter returns cartesian product of the input iterables.

Reviewer: Can you add a link to https://docs.python.org/3/library/itertools.html#itertools.product, so people who read this and think, "What on earth is a cartesian product?" have a reference?

Reviewer: I turned this comment into a suggestion . . . see below.

Also see the :ref:`zip_filter`

Product Filters
```````````````

> The product filter returns cartesian product of the input iterables.

Reviewer: ```suggestion
The product filter returns the `cartesian product <https://docs.python.org/3/library/itertools.html#itertools.product>`_ of the input iterables.
```

Author: @acozine Done.

    {{ myvar | type_debug }}

Computer Therory Assertions
```````````````````````````

> There are fuctions human_readable and human_to_bytes.

Reviewer 2: ```suggestion
There are functions ``human_readable`` and ``human_to_bytes``.
```

Author: Updated.

> There are functions ``human_readable`` and ``human_to_bytes``.

Reviewer: Could you add a use case here? I'm not sure I understand these features completely, so this is a guess, but something like:

"The functions `human_readable` and `human_to_bytes` let you test your playbooks to make sure you are using the right size format in your tasks - that you're providing Byte format to computers and human-readable format to people."

Author: Sure I will add them.

Human Readable
``````````````

> This will assert given string is human readable or not.

Reviewer 2: ```suggestion
This filter will assert if the given string is human readable or not.
```

Author: Updated.

> This filter will assert if the given string is human readable or not.

Reviewer: ```suggestion
Asserts whether the given string is human readable or not.
```

For example::

  - name: "Human Readable"
    assert:
      that:
        - '"1.00 Bytes" == 1|human_readable'
        - '"1.00 bits" == 1|human_readable(isbits=True)'
        - '"10.00 KB" == 10240|human_readable'
        - '"97.66 MB" == 102400000|human_readable'
        - '"0.10 GB" == 102400000|human_readable(unit="G")'
        - '"0.10 Gb" == 102400000|human_readable(isbits=True, unit="G")'

This would result in::

    { "changed": false, "msg": "All assertions passed" }

Human to Bytes
``````````````

> This will assert given string is in the Bytes format or not.

Reviewer 2: ```suggestion
This filter will return the given string in the Bytes format.
```

Author: Updated.

> This filter will return the given string in the Bytes format.

Reviewer: ```suggestion
Returns the given string in the Bytes format.
```

    {{ myvar | type_debug }}

> Computer Therory Assertions

Reviewer: ```suggestion
Computer Theory Assertions
```

Reviewer: oops, missed a typo

```````````````````````````

> There are functions ``human_readable`` and ``human_to_bytes`` let you test your 

Reviewer: ```suggestion
The ``human_readable`` and ``human_to_bytes`` functions let you test your 
```

Author: @acozine Done. Anything need to add or change .? Thanks.

Reviewer: @sbulage thanks for all your work on this PR! One last detail - you missed the second part of this change - adding "functions" before the word "let", so the line reads:

"The `human_readable` and `human_to_bytes` **functions** let you test . . ."

Author: No worries!!! I will update it.

Author: @acozine Done.

Reviewer: Looks good. As soon as Shippable (CI) passes, I will merge. Thanks again @sbulage.

Author: Thanks @acozine  :)
